zebra-consensus/src/transaction.rs

```diff
@@ -518,31 +499,66 @@ where
                 //.map_err(|e| BoxError::from(Box::new(e)))?;
             }
 
-            let _rsp = redjubjub_verifier
-                .ready_and()
-                .await?
-                .call((bvk, sapling_shielded_data.binding_sig, &shielded_sighash).into())
-                .boxed();
-
             // TODO: stop ignoring binding signature errors - #1939
-            // async_checks.push(rsp);
+            // async_checks.include(
+            //     primitives::redjubjub::VERIFIER
+            //         .clone()
+            //         .oneshot((bvk, sapling_shielded_data.binding_sig, &shielded_sighash).into()),
+            // );
         }
 
         Ok(async_checks)
     }
+}
+
+/// A set of asynchronous checks that should succeed.
+///
+/// A wrapper around [`FuturesUnordered`] with some auxiliary methods.
+struct AsyncChecks(FuturesUnordered<Pin<Box<dyn Future<Output = Result<(), BoxError>> + Send>>>);
+
+impl AsyncChecks {
+    /// Create an empty set of asynchronous checks.
+    pub fn new() -> Self {
+        AsyncChecks(FuturesUnordered::new())
+    }
+
+    /// Include a check in the set.
+    pub fn include(&mut self, check: impl Future<Output = Result<(), BoxError>> + Send + 'static) {
+        self.0.push(check.boxed());
+    }
 
-    /// Await a set of checks that should all succeed.
+    /// Include a set of checks in the set.
+    ///
+    /// This method can be daisy-chained.
+    pub fn and(mut self, checks: AsyncChecks) -> Self {
+        self.0.extend(checks.0);
+        self
+    }
+
+    /// Wait until all checks in the set finish.
```

Comment:
Missing documentation: Let's explicitly say that the checks are unordered.

Reply:
Done in [c340478](https://github.com/jvff/zebra/commit/c340478bef309061e1e9752d464ffe7aba8aabe9).